src/Utilities/Tags.jl

```diff
@@ -0,0 +1,120 @@
+module Tags
+
+using Compat
+
+const BLOCK_ELEMENTS = Set([
+    :address, :article, :aside, :blockquote, :canvas, :dd, :div, :dl,
+    :fieldset, :figcaption, :figure, :footer, :form, :h1, :h2, :h3, :h4, :h5,
+    :h6, :header, :hgroup, :hr, :li, :main, :nav, :noscript, :ol, :output, :p,
+    :pre, :section, :table, :tfoot, :ul, :video,
+])
+const INLINE_ELEMENTS = Set([
+    :a, :abbr, :acronym, :b, :bdo, :big, :br, :button, :cite, :code, :dfn, :em,
+    :i, :img, :input, :kbd, :label, :map, :object, :q, :samp, :script, :select,
+    :small, :span, :strong, :sub, :sup, :textarea, :time, :tt, :var,
+])
+const VOID_ELEMENTS = Set([
+    :area, :base, :br, :col, :command, :embed, :hr, :img, :input, :keygen,
+    :link, :meta, :param, :source, :track, :wbr,
+])
+const ALL_ELEMENTS = union(BLOCK_ELEMENTS, INLINE_ELEMENTS, VOID_ELEMENTS)
+
+const EMPTY_STRING = ""
+const TEXT = Symbol(EMPTY_STRING)
+
+immutable Tag
+    name :: Symbol
+end
+
+Base.show(io::IO, t::Tag) = print(io, "<", t.name, ">")
+
+macro tags(args...) esc(tags(args)) end
+tags(s) = :(const ($(s...),) = $(map(Tag, s)))
+
+typealias Attributes Vector{Pair{Symbol, String}}
+
+immutable Node
+    name :: Symbol
+    text :: String
+    attr :: Attributes
+    data :: Vector{Node}
+
+    Node(name::Symbol, attr::Attributes, data::Vector{Node}) = new(name, EMPTY_STRING, attr, data)
+    Node(name::Symbol, text::AbstractString) = (@assert(name === TEXT); new(name, text))
```

Comment:
An option here would also be `Node(text::AbstractString) = new(TEXT,text)` to easily create text nodes. It shouldn't clash, but perhaps it is better to be explicit.